addons/hr_payroll/models/hr_contract.py

```diff
@@ -0,0 +1,36 @@
+#-*- coding:utf-8 -*-
+# Part of Odoo. See LICENSE file for full copyright and licensing details.
+
+from odoo import api, fields, models
+
+
+class HrContract(models.Model):
+    """
+    Employee contract based on the visa, work permits
+    allows to configure different Salary structure
+    """
+
+    _inherit = 'hr.contract'
+    _description = 'Employee Contract'
+
+    struct_id = fields.Many2one('hr.payroll.structure', 'Salary Structure')
+    schedule_pay = fields.Selection([
+        ('monthly', 'Monthly'),
+        ('quarterly', 'Quarterly'),
+        ('semi-annually', 'Semi-annually'),
+        ('annually', 'Annually'),
+        ('weekly', 'Weekly'),
+        ('bi-weekly', 'Bi-weekly'),
+        ('bi-monthly', 'Bi-monthly'),
+    ], 'Scheduled Pay', index=True, default='monthly')
+
+    @api.multi
+    def get_all_structures(self, contract_ids):
```

Comment:
contract_ids is current object ids. is it required to pass as argument ? no need to pass contract_ids , self is enough and  call get_all_structures with recordset..